src/main/java/org/generationcp/middleware/dao/LocationDAO.java

```diff
@@ -300,43 +301,72 @@ public long countByCountry(final Country country) {
 		return new ArrayList<>();
 	}
 
-	public List<Location> filterLocations(final String programUUID, final Set<Integer> types, final List<Integer> locationIds,
-		final List<String> locationAbbreviations) {
-		final List<Location> locations;
+	public long countFilterLocations(final String programUUID, final Set<Integer> types, final List<Integer> locationIds,
```

Comment:
Let's have a `LocationSearchRequest` class with all the possible search parameters. This way we don't end up having to modify numerous methods of DAOs, services whenever modifying/adding possible location search parameters